lib/opal/nodes/top.rb

```diff
@@ -108,6 +172,66 @@ def compile_end_construct
       def version_comment
         "/* Generated by Opal #{Opal::VERSION} */"
       end
+
+
+      def self.module_name(filenamepath, original_filepath = nil, original_filename = nil)
+        original_filename = filenamepath unless original_filename
+        original_filepath = Pathname.new(filenamepath).expand_path unless original_filepath
+        o_s = original_filepath.to_s
+        path, _ = Pathname.new(filenamepath).expand_path.split
+        if Opal.paths.include?(path.expand_path.to_s)
+          e = if o_s.end_with?('.js.rb')
+                -7
+              elsif o_s.end_with?('.rb') || o_s.end_with?('.js')
+                -4
+              else
+                -1
+              end
+          return o_s[(path.expand_path.to_s.size+1)..e]
+        end
+        if path.root?
+          pwd = Dir.pwd
+          if o_s.start_with?(pwd)
+            e = if o_s.end_with?('.js.rb')
```

Comment:
Same here, `String#sub` + there seems to be a code duplication. Probably it would be easier to extract to a separate method (in a separate class that probably should be created)